designpatterns/gangoffour/src/main/kotlin/com/felipecosta/designpatterns/creational/Builder.kt

```diff
@@ -0,0 +1,109 @@
+package com.felipecosta.designpatterns.creational
+
+//awful repetition, but this is to avoid clashing with other patterns - I'm open to sugestions.
+
+abstract class MazeBuilder {
+
+    internal val mazeBeingBuilt = BuilderMaze();
+
+    abstract fun makeRoom(number: Int): BuilderRoom;
+    abstract fun makeMazeWall(): BuilderWall;
+    abstract fun makeDoor(outRoom: BuilderRoom, inRoom: BuilderRoom): BuilderDoor;
+
+    fun getResult(): BuilderMaze {
+        return mazeBeingBuilt;
+    }
+}
+
+open class MazeDirector(val builder: MazeBuilder) {
+    fun Construct() {
```

Comment:
Sorry about that! Coding too much C++ and reading from a C++ book can get confusing at times!
I will fix this ASAP.